src/librustc_trans/attributes.rs

```diff
@@ -75,9 +76,17 @@ pub fn from_fn_attrs(ccx: &CrateContext, attrs: &[ast::Attribute], llfn: ValueRe
     inline(llfn, find_inline_attr(Some(ccx.sess().diagnostic()), attrs));
 
     set_frame_pointer_elimination(ccx, llfn);
-
+    let mut target_features = vec![];
     for attr in attrs {
-        if attr.check_name("cold") {
+        if attr.check_name("target_feature") {
+            let val = attr.value_str().map_or(String::new(), |s|s.to_string());
```

Comment:
It turns out this doesn't work because `s` is an `InternedString`. I just took the hit with the extra case analysis instead.